src/components/Slices/SliceDetails.jsx

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable react/no-array-index-key */
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  Typography,
+  Grid,
+  Tabs,
+  Tab,
+  Box,
+  Tooltip,
+  AppBar,
+  makeStyles,
+} from "@material-ui/core";
+
+import UTXOSet from "../ScriptExplorer/UTXOSet";
+import MultisigDetails from "../MultisigDetails";
+import ImportAddressesButton from "../ImportAddressesButton";
+import ConfirmAddress from "./ConfirmAddress";
+
+import { slicePropTypes, clientPropTypes } from "../../proptypes";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+    backgroundColor: theme.palette.background.paper,
+  },
+}));
+
+function a11yProps(index) {
+  return {
+    id: `scrollable-auto-tab-${index}`,
+    "aria-controls": `scrollable-auto-tabpanel-${index}`,
+  };
+}
+
+const createTab = ({ tab }, index) => {
+  const tooltipTab = (
+    <Tab value={index.toString()} key={index} {...a11yProps(index)} {...tab} />
+  );
+  if (tab.disabled)
+    return (
+      <Tooltip title="Not available for this address" key={index} arrow>
+        <div>{tooltipTab}</div>
+      </Tooltip>
+    );
+  return tooltipTab;
+};
+
+function TabPanel({ children, value, index, ...other }) {
+  return (
+    <Grid
+      container
+      justify="center"
+      style={{ display: value === index ? "inherit" : "none" }}
+      {...other}
+    >
+      <Grid item md={11}>
+        <Box>{children}</Box>
+      </Grid>
+    </Grid>
+  );
+}
+
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  value: PropTypes.string.isRequired,
+  index: PropTypes.string.isRequired,
+};
+
+TabPanel.defaultProps = {
+  children: React.createElement("span"),
+};
+
+const SliceDetails = ({ slice, client, network }) => {
```

Comment:
This is the "Detail Panel" that expands under a row in the slices table. It doesn't need much information passed down to it to render all of the slice detail views a user could want, the most important thing being the slice data (as defined in our shared proptype now).